app/utils/supabase.ts

```diff
@@ -0,0 +1,434 @@
+ * are available.
+ *
+ * @param options - Optional configuration for the fetch operation
+ * @param options.limit - Maximum number of resources to fetch (default: 50)
+ * @param options.offset - Number of resources to skip for pagination (default: 0)
+ * @param options.resourceType - Filter resources by type (WATER, FOOD, FORAGE, BATHROOM)
+ * @param options.status - Filter resources by status (OPERATIONAL, TEMPORARILY_CLOSED, etc.)
+ * @returns Promise resolving to an object containing the data, count, and hasMore flag
+ * @throws Error if the database query fails
+ *
+ * @example
+ * // Fetch first 10 resources
+ * const result = await getResources({ limit: 10, offset: 0 });
+ * console.log(`Fetched ${result.data.length} of ${result.count} total resources`);
+ * if (result.hasMore) {
+ *   console.log('More resources available');
+ * }
+ *
+ * @example
+ * // Fetch only operational water resources
+ * const waterResources = await getResources({
+ *   resourceType: 'WATER',
+ *   status: 'OPERATIONAL',
+ *   limit: 100
+ * });
+ *
+ * @example
+ * // Paginate through all resources
+ * let offset = 0;
+ * const limit = 50;
+ * let allResources: ResourceEntry[] = [];
+ *
+ * while (true) {
+ *   const result = await getResources({ limit, offset });
+ *   allResources = [...allResources, ...result.data];
+ *   if (!result.hasMore) break;
+ *   offset += limit;
+ * }
+ */
+export async function getResources(
+  options: FetchResourcesOptions = {}
+): Promise<FetchResourcesResult> {
+  const { limit = 50, offset = 0, resourceType, status } = options;
+
+  // Build the query with filters
+  let query = supabase
+    .from(resourceDatabaseName)
+    .select('*', { count: 'exact' })
+    .range(offset, offset + limit - 1);
+
+  // Apply filters if provided
+  if (resourceType) {
+    query = query.eq('resource_type', resourceType);
+  }
+  if (status) {
+    query = query.eq('status', status);
+  }
+
+  const { data, error, count } = await query;
+
+  if (error) {
+    throw new Error(`Failed to fetch resources: ${error.message}`);
+  }
+
+  return {
+    data: (data || []) as ResourceEntry[],
+    count,
+    hasMore: count !== null && offset + limit < count,
+  };
+}
+
+/**
+ * Fetches a single resource by its ID.
+ *
+ * @param id - The unique identifier of the resource
+ * @returns Promise resolving to the resource entry, or null if not found
+ * @throws Error if the database query fails
+ *
+ * @example
+ * const resource = await getResourceById('123e4567-e89b-12d3-a456-426614174000');
+ * if (resource) {
+ *   console.log(`Found resource: ${resource.name}`);
+ * } else {
+ *   console.log('Resource not found');
+ * }
+ */
+export async function getResourceById(
+  id: string
+): Promise<ResourceEntry | null> {
+  const { data, error } = await supabase
+    .from(resourceDatabaseName)
+    .select('*')
+    .eq('id', id)
+    .single();
+
+  if (error) {
+    if (error.code === 'PGRST116') {
```

Comment:
Does supabase provides enums for these error codes? Can we have a more descriptive and human-readable mapping of these errors?